### ISSUE TITLE
etcd member fails to join the cluster after IPsec upgrade

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#10103]
**Rancher versions:**
rancher/server:v1.6.10 to v1.6.11-rc3

**Docker version: (`docker version`,`docker info` preferred)**1.12.6

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**Ubuntu

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**Digital OCean

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)** Single node

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**Kubernetes Version: v1.7.7-rancher1

**Steps to Reproduce:**
1. Create a 3 host Kubernetes setup in v1.6.10. Create pods/rcs
2. Upgrade rancher to v1.6.11-rc3. Verify the pods and rcs are functional 
3. Upgrade Network services stack. Create more pods/rcs
4. Upgrade IPsec stack

**Results:**
After IPsec stack upgrade, one of the etcd members fails to join the cluster. The etcd cannot talk to other etcd members.
etcd cluster-health from the member which fails to join the cluster. 
```
bash-4.3# etcdctl cluster-health
failed to check the health of member 2b0ab6faf0dd5554 on http://10.42.93.129:237
9: Get http://10.42.93.129:2379/health: dial tcp 10.42.93.129:2379: i/o timeout
member 2b0ab6faf0dd5554 is unreachable: [http://10.42.93.129:2379] are all unrea
chable
failed to check the health of member 803085876e049bbb on http://10.42.219.169:23
79: Get http://10.42.219.169:2379/health: dial tcp 10.42.219.169:2379: i/o timeo
ut
member 803085876e049bbb is unreachable: [http://10.42.219.169:2379] are all unre
achable
member c191fc353e1eaa51 is unhealthy: got unhealthy result from http://10.42.183
.201:2379
cluster is unhealthy
```
Creation of new pod fails
```
Error from server (InternalError): error when creating "pod6.yml": Internal error occurred: failed to allocate a service
IP: client: etcd cluster is unavailable or misconfigured; error #0: client: etcd member http://etcd.kubernetes.rancher.i
nternal:2379 has no leader

Error from server: error when creating "pod6.yml": client: etcd member http://etcd.kubernetes.rancher.internal:2379 has
no leader

```
Addon-starter fails to start. 

Cluster health from other etcd members is healthy. The other etcd members can talk to each other.

**Comment by leodotcloud** [comment]
etcd:
```
1i75      kubernetes-etcd-1                                                rancher/etcd:v2.3.7-13                                                                                                         healthy        1h4       10.42.219.169   0e36f83b5901
1i94      kubernetes-etcd-2                                                rancher/etcd:v2.3.7-13                                                                                                         healthy        1h5       10.42.93.129    0c266cb8d19d
1i96      kubernetes-etcd-3                                                rancher/etcd:v2.3.7-13                                                                                                         healthy        1h6       10.42.183.201   fc8157e6bb55
1i1141    leoetcd                                                          leodotcloud/swiss-army-knife:dev                                                                                               running        1h6                       21a07b9ba0d1
```

ipsec:
```
1i205     ipsec-ipsec-2                                                    rancher/net:holder                                                                                                             running        1h4       10.42.79.194    f80e01c13a0e
1i217     ipsec-ipsec-1                                                    rancher/net:holder                                                                                                             running        1h6       10.42.47.190    7b63d75e10b9
1i221     ipsec-ipsec-3                                                    rancher/net:holder                                                                                                             running        1h5       10.42.87.244    727d3bb0c098
```

kubernetes-etcd-3 and ipsec-ipsec-1 are running on the same host: 1h6

Inside: kubernetes-etcd-3
```
root@fc8157e6bb55:/# ip neigh | grep 10.42.219.169
10.42.219.169 dev eth0 lladdr 02:32:56:3b:a1:9d REACHABLE
root@fc8157e6bb55:/#
```

Looking at ipsec-ipsec-3:
```
root@7b63d75e10b9:/# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
3: eth0@if36: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
    link/ether 02:32:56:f8:bd:15 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.42.47.190/16 brd 10.42.255.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::2467:86ff:fe25:eeab/64 scope link
       valid_lft forever preferred_lft forever
root@7b63d75e10b9:/#
```

The ARP entry for `10.42.219.169` should be pointing to mac address of ipsec-ipsec-3 (`02:32:56:f8:bd:15`), instead it's pointing to `02:32:56:3b:a1:9d`. Not sure whose mac address it is.

Checking for dangling veths, I couldn't find any. All veths have been accounted for and there are no extra ones.
```
10: 01785df@if9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
12: 97f031a8fd: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
97f031a8fd41        rancher/healthcheck:v0.3.3                                                                                                    "/.r/r /rancher-entry"   26 hours ago         Up 26 hours                             r-healthcheck-healthcheck-1-29395c8a
782: 9086d6789a: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
9086d6789a25        rancher/k8s:v1.7.7-rancher1                                                                                                   "/.r/r /usr/bin/entry"   26 hours ago         Up 13 hours                             r-kubernetes-controller-manager-1-39e78075
18: 3d8ffa311f: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
3d8ffa311f42        rancher/k8s:v1.7.7-rancher1                                                                                                   "/.r/r /usr/bin/entry"   26 hours ago         Up 26 hours                             r-kubernetes-kubernetes-1-7a73b167
20: fc8157e6bb: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
fc8157e6bb55        rancher/etcd:v2.3.7-13                                                                                                        "/.r/r /opt/rancher/r"   26 hours ago         Up 26 hours                             r-kubernetes-etcd-3-0b1bd6d6
23: 0763d286c1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
0763d286c1d8        gcr.io/google_containers/pause-amd64:3.0                                                                                      "/pause"                 26 hours ago         Up 26 hours                             k8s_POD_tiller-deploy-737598192-1566c_kube-system_2c4a01b0-af7d-11e7-8461-023256ad13e5_0
24: 394a2d50d7: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
394a2d50d726        gcr.io/google_containers/pause-amd64:3.0                                                                                      "/pause"                 26 hours ago         Up 26 hours                             k8s_POD_kubernetes-dashboard-716739405-vnws5_kube-system_2c4d6ee2-af7d-11e7-8461-023256ad13e5_0
25: eb9d819366: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
eb9d819366bb        gcr.io/google_containers/pause-amd64:3.0                                                                                      "/pause"                 26 hours ago         Up 26 hours                             k8s_POD_k8testrc1-8rwdt_default_93928d4a-af7d-11e7-8461-023256ad13e5_0
26: 924d606b03: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
924d606b0385        gcr.io/google_containers/pause-amd64:3.0                                                                                      "/pause"                 26 hours ago         Up 26 hours                             k8s_POD_nginx-x23jp_default_ae3d4dbb-af7d-11e7-8461-023256ad13e5_0
32: 2d915db62a: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
2d915db62acf        gcr.io/google_containers/pause-amd64:3.0                                                                                      "/pause"                 26 hours ago         Up 26 hours                             k8s_POD_k8testrc1-61fpb_default_21715910-af7f-11e7-8461-023256ad13e5_0
33: 5e9946771b: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
5e9946771bc6        gcr.io/google_containers/pause-amd64:3.0                                                                                      "/pause"                 26 hours ago         Up 26 hours                             k8s_POD_k8testrc3-swj6r_default_3215ffd9-af7f-11e7-8461-023256ad13e5_0
34: 976ae544ab: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
976ae544ab26        rancher/lb-service-haproxy:v0.7.13                                                                                            "/.r/r /tini -- lb-co"   26 hours ago         Up 26 hours                             r-kubernetes-ingress-lbs-default-rancherlb-ingress3-1-642459bb
35: c7e879efd9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
c7e879efd9df        rancher/kubernetes-auth:v0.0.8                                                                                                "/.r/r /usr/bin/entry"   26 hours ago         Up 26 hours                             r-kubernetes-rancher-kubernetes-auth-1-97b93a9e
36: 7b63d75e10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
7b63d75e10b9        rancher/net:holder                                                                                                            "/.r/r /rancher-entry"   26 hours ago         Up 26 hours                             r-ipsec-ipsec-1-a79814b7
1632: 9feba1d1cd: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
9feba1d1cd34        rancher/k8s:v1.7.7-rancher1                                                                                                   "/.r/r /usr/bin/entry"   About a minute ago   Up 6 seconds                            r-kubernetes-addon-starter-1-931110ec
1485: 1ee8aaea6a: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
1ee8aaea6a54        rancher/kubernetes-agent:v0.6.6                                                                                               "/.r/r /usr/bin/entry"   26 hours ago         Up 2 hours                              r-kubernetes-rancher-kubernetes-agent-1-3c27d808
CONTAINER ID        IMAGE                                                                                                                         COMMAND                  CREATED              STATUS              PORTS               NAMES
21a07b9ba0d1        leodotcloud/swiss-army-knife:dev                                                                                              "/bin/sh -c /usr/bin/"   9 minutes ago        Up 9 minutes                            leoetcd
9a47b62fbdae        rancher/net:v0.13.0                                                                                                           "/rancher-entrypoint."   26 hours ago         Up 26 hours                             r-ipsec-ipsec-router-1-24dd1c5c
165989a1cf26        rancher/net:v0.13.0                                                                                                           "/rancher-entrypoint."   26 hours ago         Up 26 hours                             r-ipsec-cni-driver-1-f8e94c56
89b87d156e4f        rancher/network-manager:v0.7.11                                                                                               "/rancher-entrypoint."   26 hours ago         Up 26 hours                             r-network-services-network-manager-1-febdf4e8
58c33e9b5427        sangeetha/testnewhostrouting@sha256:07f6b52d5c67c71dea08ccec41d7f09083da1c3e9c2e3932d2f1ef00afbf77af                          "/scripts/run.sh ngin"   26 hours ago         Up 26 hours                             k8s_testcontainer_k8testrc3-swj6r_default_3215ffd9-af7f-11e7-8461-023256ad13e5_0
bf094f525e68        sangeetha/testnewhostrouting@sha256:07f6b52d5c67c71dea08ccec41d7f09083da1c3e9c2e3932d2f1ef00afbf77af                          "/scripts/run.sh ngin"   26 hours ago         Up 26 hours                             k8s_testcontainer_k8testrc1-61fpb_default_21715910-af7f-11e7-8461-023256ad13e5_0
32f8296762ae        rancher/agent:v1.2.7-rc2                                                                                                      "/run.sh run"            26 hours ago         Up 26 hours                             rancher-agent
4f8e0186fd6e        sangeetha/testnewhostrouting@sha256:07f6b52d5c67c71dea08ccec41d7f09083da1c3e9c2e3932d2f1ef00afbf77af                          "/scripts/run.sh ngin"   26 hours ago         Up 26 hours                             k8s_nginx_nginx-x23jp_default_ae3d4dbb-af7d-11e7-8461-023256ad13e5_0
98286a05e689        sangeetha/testnewhostrouting@sha256:07f6b52d5c67c71dea08ccec41d7f09083da1c3e9c2e3932d2f1ef00afbf77af                          "/scripts/run.sh ngin"   26 hours ago         Up 26 hours                             k8s_testcontainer_k8testrc1-8rwdt_default_93928d4a-af7d-11e7-8461-023256ad13e5_0
6724583e5cf9        gcr.io/google_containers/kubernetes-dashboard-amd64@sha256:b537ce8988510607e95b8d40ac9824523b1f9029e6f9f90e9fccc663c355cf5d   "/dashboard --insecur"   26 hours ago         Up 26 hours                             k8s_kubernetes-dashboard_kubernetes-dashboard-716739405-vnws5_kube-system_2c4d6ee2-af7d-11e7-8461-023256ad13e5_0
9a218a9115a0        gcr.io/kubernetes-helm/tiller@sha256:fc5bef1464e78abe683c3e9e5e52dcb89111ab129049f8b3819eb157716fe4f7                         "/tiller"                26 hours ago         Up 26 hours                             k8s_tiller_tiller-deploy-737598192-1566c_kube-system_2c4a01b0-af7d-11e7-8461-023256ad13e5_0
af5bee09a716        rancher/etc-host-updater:v0.0.3                                                                                               "/rancher-entrypoint."   26 hours ago         Up 26 hours                             r-kubernetes-kubernetes-kube-hostname-updater-1-bc7556a2
84da1d5be010        rancher/k8s:v1.7.7-rancher1                                                                                                   "/usr/bin/entry.sh ku"   26 hours ago         Up 26 hours                             r-kubernetes-proxy-1-e2179441
53dbb29bf723        rancher/k8s:v1.7.7-rancher1                                                                                                   "/usr/bin/entry.sh ku"   26 hours ago         Up 26 hours                             r-kubernetes-kubelet-1-c1110383
1c8969fe23fe        rancher/dns:v0.15.3                                                                                                           "/rancher-entrypoint."   26 hours ago         Up 26 hours                             r-network-services-metadata-dns-1-ef162cca
1df00d6e3456        rancher/metadata:v0.9.4                                                                                                       "/rancher-entrypoint."   26 hours ago         Up 26 hours                             r-network-services-metadata-1-40ff1c18
total 0
```

Going through all the network namespaces on the proc filesystem and searching for that MAC address didn't reveal anything.
```
namespaces=$(find . -name "net*" | grep "ns/net" | grep -v task)
for aNS in $namespaces; do nsenter --net=$aNS ip addr; done | grep "link/ether" >> /tmp/list_of_macs_netns.txt
root@soumyaadkubetest3:/tmp# cat list_of_macs_netns.txt | grep 02:32:56:3b:a1:9d
root@soumyaadkubetest3:/tmp# echo $?
1
root@soumyaadkubetest3:/tmp#
```

Looking through the cattle db, couldn't figure out which instance this mac address was associated with in the past. Issue: https://github.com/rancher/rancher/issues/9516
```
mysql> select * from nic where mac_address="02:32:56:3b:a1:9d";
Empty set (0.00 sec)

mysql>
```

**Comment by leodotcloud** [comment]
Thanks to @alena1108, looks like there is information about this in instance table under data field.

```
mysql> select * from instance where data like "%02:32:56:3b:a1:9d%" \G
*************************** 1. row ***************************
                     id: 61
                   name: ipsec-ipsec-1
             account_id: 20
                   kind: container
                   uuid: da15394a-a929-413a-8667-7365d17706f9
            description: NULL
                  state: purged
                created: 2017-10-12 18:41:37
                removed: 2017-10-12 19:23:59
            remove_time: 2017-10-12 19:26:04
                   data: {"fields":{"networkMode":"ipsec","serviceIndex":"1","logConfig":{},"internalVolumes":[],"publishAllPorts":false,"vcpu":1,"runInit":false,"stdinOpen":false,"readOnly":false,"ports":["500:500/udp","4500:4500/udp"],"allocatedIpAddress":null,"command":["sh","-c","echo Refer to router sidekick for logs; mkfifo f; exec cat f"],"labels":{"io.rancher.service.deployment.unit":"1a2fc239-8168-412d-a3a9-4814919f8065","io.rancher.cni.link_mtu_overhead":"0","io.rancher.network.arpsync":"true","io.rancher.network.macsync":"true","io.rancher.scheduler.global":"true","io.rancher.service.requested.host.id":"6","io.rancher.sidekicks":"router","io.rancher.stack_service.name":"ipsec/ipsec","io.rancher.service.launch.config":"io.rancher.service.primary.launch.config","io.rancher.project.name":"ipsec","io.rancher.project_service.name":"ipsec/ipsec","io.rancher.stack.name":"ipsec","io.rancher.service.hash":"66e5a4d058439e6fc53f0d8f3c2272dc2ea8343a","io.rancher.container.system":"true","io.rancher.environment.uuid":"77fcb4ce-1778-4fb5-8e81-c472100acfa3","io.rancher.container.ip":"10.42.195.209/16","io.rancher.container.uuid":"da15394a-a929-413a-8667-7365d17706f9","io.rancher.cni.network":"ipsec","io.rancher.cni.wait":"true","io.rancher.container.mac_address":"02:32:56:3b:a1:9d","io.rancher.container.name":"ipsec-ipsec-1"},"privileged":false,"startOnCreate":true,"dataVolumeMounts":{},"tty":false,"dnsSearch":["ipsec.rancher.internal","ipsec.ipsec.rancher.internal","rancher.internal"],"requestedHostId":"6","imageUuid":"docker:rancher/net:holder","networkIds":[10],"dns":["169.254.169.250"],"dataVolumes":[],"transitioningProgress":100,"hostId":6,"dockerHostIp":"165.227.102.15","dockerIp":"","dockerPorts":[],"userPorts":["500:500/udp","4500:4500/udp"],"primaryIpAddress":"10.42.195.209"},"dockerContainer":{"Id":"de44e949d96f451d2ba247eefb897024d6207e182e2e67061a62537d54258d59","Names":["/r-ipsec-ipsec-1-da15394a"],"Image":"rancher/net:holder","ImageID":"sha256:665d9f6e8cc1ec3fa1181398d0661715cf71134db65666072998c54dbeeb7072","Command":"/.r/r /rancher-entrypoint.sh sh -c 'echo Refer to router sidekick for logs; mkfifo f; exec cat f'","Created":1507833719,"Ports":[],"Labels":{"io.rancher.cni.link_mtu_overhead":"0","io.rancher.cni.network":"ipsec","io.rancher.cni.wait":"true","io.rancher.container.ip":"10.42.195.209/16","io.rancher.container.mac_address":"02:32:56:3b:a1:9d","io.rancher.container.name":"ipsec-ipsec-1","io.rancher.container.system":"true","io.rancher.container.uuid":"da15394a-a929-413a-8667-7365d17706f9","io.rancher.environment.uuid":"77fcb4ce-1778-4fb5-8e81-c472100acfa3","io.rancher.network.arpsync":"true","io.rancher.network.macsync":"true","io.rancher.project.name":"ipsec","io.rancher.project_service.name":"ipsec/ipsec","io.rancher.scheduler.global":"true","io.rancher.service.deployment.unit":"1a2fc239-8168-412d-a3a9-4814919f8065","io.rancher.service.hash":"66e5a4d058439e6fc53f0d8f3c2272dc2ea8343a","io.rancher.service.launch.config":"io.rancher.service.primary.launch.config","io.rancher.service.requested.host.id":"6","io.rancher.sidekicks":"router","io.rancher.stack.name":"ipsec","io.rancher.stack_service.name":"ipsec/ipsec"},"State":"exited","Status":"Exited (137) Less than a second ago","HostConfig":{"NetworkMode":"none"},"NetworkSettings":{"Networks":{"none":{"IPAMConfig":null,"Links":null,"Aliases":null,"NetworkID":"2848f433a311c826572fec51f28bb6b94a2fb7dc1e47e6f10dce934779656a7c","EndpointID":"","Gateway":"","IPAddress":"","IPPrefixLen":0,"IPv6Gateway":"","GlobalIPv6Address":"","GlobalIPv6PrefixLen":0,"MacAddress":""}}},"Mounts":[{"Name":"rancher-cni","Source":"/var/lib/docker/volumes/rancher-cni/_data","Destination":"/.r","Driver":"local","Mode":"ro","RW":false,"Propagation":"rprivate"}]},"dockerInspect":{"Id":"de44e949d96f451d2ba247eefb897024d6207e182e2e67061a62537d54258d59","Created":"2017-10-12T18:41:59.072043568Z","Path":"/.r/r","Args":["/rancher-entrypoint.sh","sh","-c","echo Refer to router sidekick for logs; mkfifo f; exec cat f"],"State":{"Status":"exited","Running":false,"Paused":false,"Restarting":false,"OOMKilled":false,"Dead":false,"Pid":0,"ExitCode":137,"Error":"","StartedAt":"2017-10-12T18:42:02.031513544Z","FinishedAt":"2017-10-12T19:22:41.174328486Z"},"Image":"sha256:665d9f6e8cc1ec3fa1181398d0661715cf71134db65666072998c54dbeeb7072","ResolvConfPath":"/var/lib/docker/containers/de44e949d96f451d2ba247eefb897024d6207e182e2e67061a62537d54258d59/resolv.conf","HostnamePath":"/var/lib/docker/containers/de44e949d96f451d2ba247eefb897024d6207e182e2e67061a62537d54258d59/hostname","HostsPath":"/var/lib/docker/containers/de44e949d96f451d2ba247eefb897024d6207e182e2e67061a62537d54258d59/hosts","LogPath":"/var/lib/docker/containers/de44e949d96f451d2ba247eefb897024d6207e182e2e67061a62537d54258d59/de44e949d96f451d2ba247eefb897024d6207e182e2e67061a62537d54258d59-json.log","Name":"/r-ipsec-ipsec-1-da15394a","RestartCount":0,"Driver":"aufs","MountLabel":"","ProcessLabel":"","AppArmorProfile":"","ExecIDs":null,"HostConfig":{"Binds":["rancher-cni:/.r:ro"],"ContainerIDFile":"","LogConfig":{"Type":"json-file","Config":{}},"NetworkMode":"none","PortBindings":{},"RestartPolicy":{"Name":"","MaximumRetryCount":0},"AutoRemove":false,"VolumeDriver":"","VolumesFrom":null,"CapAdd":null,"CapDrop":null,"Dns":["169.254.169.250"],"DnsOptions":null,"DnsSearch":["ipsec.rancher.internal","ipsec.ipsec.rancher.internal","rancher.internal"],"ExtraHosts":null,"GroupAdd":null,"IpcMode":"","Cgroup":"","Links":null,"OomScoreAdj":0,"PidMode":"","Privileged":false,"PublishAllPorts":false,"ReadonlyRootfs":false,"SecurityOpt":null,"UTSMode":"","UsernsMode":"","ShmSize":67108864,"Runtime":"runc","ConsoleSize":[0,0],"Isolation":"","CpuShares":2,"Memory":0,"CgroupParent":"","BlkioWeight":0,"BlkioWeightDevice":null,"BlkioDeviceReadBps":null,"BlkioDeviceWriteBps":null,"BlkioDeviceReadIOps":null,"BlkioDeviceWriteIOps":null,"CpuPeriod":0,"CpuQuota":0,"CpuRealtimePeriod":0,"CpuRealtimeRuntime":0,"CpusetCpus":"","CpusetMems":"","Devices":[],"DiskQuota":0,"KernelMemory":0,"MemoryReservation":0,"MemorySwap":0,"MemorySwappiness":-1,"OomKillDisable":false,"PidsLimit":0,"Ulimits":null,"CpuCount":0,"CpuPercent":0,"IOMaximumIOps":0,"IOMaximumBandwidth":0},"GraphDriver":{"Name":"aufs","Data":null},"Mounts":[{"Name":"rancher-cni","Source":"/var/lib/docker/volumes/rancher-cni/_data","Destination":"/.r","Driver":"local","Mode":"ro","RW":false,"Propagation":"rprivate"}],"Config":{"Hostname":"de44e949d96f","Domainname":"","User":"","AttachStdin":false,"AttachStdout":false,"AttachStderr":false,"Tty":false,"OpenStdin":false,"StdinOnce":false,"Env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin","SSL_SCRIPT_COMMIT=98660ada3d800f653fc1f105771b5173f9d1a019","CNI=v0.3.0","RANCHER_CNI_BRIDGE=v0.3.0","RANCHER_CNI_IPAM=v0.1.1"],"Cmd":["sh","-c","echo Refer to router sidekick for logs; mkfifo f; exec cat f"],"Healthcheck":{},"Image":"rancher/net:holder","Volumes":null,"WorkingDir":"","Entrypoint":["/.r/r","/rancher-entrypoint.sh"],"OnBuild":null,"Labels":{"io.rancher.cni.link_mtu_overhead":"0","io.rancher.cni.network":"ipsec","io.rancher.cni.wait":"true","io.rancher.container.ip":"10.42.195.209/16","io.rancher.container.mac_address":"02:32:56:3b:a1:9d","io.rancher.container.name":"ipsec-ipsec-1","io.rancher.container.system":"true","io.rancher.container.uuid":"da15394a-a929-413a-8667-7365d17706f9","io.rancher.environment.uuid":"77fcb4ce-1778-4fb5-8e81-c472100acfa3","io.rancher.network.arpsync":"true","io.rancher.network.macsync":"true","io.rancher.project.name":"ipsec","io.rancher.project_service.name":"ipsec/ipsec","io.rancher.scheduler.global":"true","io.rancher.service.deployment.unit":"1a2fc239-8168-412d-a3a9-4814919f8065","io.rancher.service.hash":"66e5a4d058439e6fc53f0d8f3c2272dc2ea8343a","io.rancher.service.launch.config":"io.rancher.service.primary.launch.config","io.rancher.service.requested.host.id":"6","io.rancher.sidekicks":"router","io.rancher.stack.name":"ipsec","io.rancher.stack_service.name":"ipsec/ipsec"}},"NetworkSettings":{"Bridge":"","SandboxID":"436969546f0e6167e19682707a74f75a5fab4e8f71e5ba3c55e1ae356cef2edd","HairpinMode":false,"LinkLocalIPv6Address":"","LinkLocalIPv6PrefixLen":0,"Ports":null,"SandboxKey":"/var/run/docker/netns/436969546f0e","SecondaryIPAddresses":null,"SecondaryIPv6Addresses":null,"EndpointID":"","Gateway":"","GlobalIPv6Address":"","GlobalIPv6PrefixLen":0,"IPAddress":"","IPPrefixLen":0,"IPv6Gateway":"","MacAddress":"","Networks":{"none":{"IPAMConfig":null,"Links":null,"Aliases":null,"NetworkID":"2848f433a311c826572fec51f28bb6b94a2fb7dc1e47e6f10dce934779656a7c","EndpointID":"","Gateway":"","IPAddress":"","IPPrefixLen":0,"IPv6Gateway":"","GlobalIPv6Address":"","GlobalIPv6PrefixLen":0,"MacAddress":""}}}},"dockerMounts":[{"Name":"rancher-cni","Source":"/var/lib/docker/volumes/rancher-cni/_data","Destination":"/.r","Driver":"local","Mode":"ro","RW":false,"Propagation":"rprivate"}]}
       allocation_state: inactive
                compute: NULL
              memory_mb: NULL
               image_id: 70
            offering_id: NULL
               hostname: NULL
                zone_id: 1
instance_triggered_stop: stop
               agent_id: NULL
                 domain: NULL
          first_running: 2017-10-12 18:42:03
                  token: EsCetAhZA0uhfYvYsBAs3ca60NXuoQdCr3jryoV0FCv3GLXFKKSuF1NsPJOSzjQzyUYKLNK6fV3cYVUx7Rg
               userdata: NULL
       system_container: NULL
 registry_credential_id: NULL
            external_id: de44e949d96f451d2ba247eefb897024d6207e182e2e67061a62537d54258d59
       native_container:
   network_container_id: NULL
           health_state: NULL
            start_count: 1
           create_index: 1
   deployment_unit_uuid: 1a2fc239-8168-412d-a3a9-4814919f8065
                version: 0
         health_updated: NULL
       service_index_id: 45
           dns_internal: 169.254.169.250
    dns_search_internal: ipsec.rancher.internal,ipsec.ipsec.rancher.internal,rancher.internal
     memory_reservation: NULL
  milli_cpu_reservation: NULL
                 system:
             service_id: NULL
         environment_id: NULL
1 row in set (0.11 sec)

mysql>
```

**Comment by deniseschannon** [comment]
@soumyalj We reverted the change in the catalog that was causing networking issues related to ipsec and upgrade testing should be re-tested.

**Comment by leodotcloud** [comment]
Host 1:
etcd-1 (Current IP: 10.42.219.169, current MAC: 02:32:56:c2:02:1b)
10.42.93.129 dev eth0 lladdr 02:32:56:ad:f1:b6 REACHABLE
10.42.10.192 dev eth0 lladdr 02:32:56:a6:b6:e2 STALE

ipsec (Current IP: 10.42.10.192, current MAC: 02:32:56:a6:b6:e2)  Past: 02:32:56:ad:f1:b6

Host 2:
etcd-2 (Current IP: 10.42.93.129, current MAC: 02:32:56:ff:f6:29)
ipsec (Current IP: 10.42.236.217, current MAC: 02:32:56:5c:1e:1c)


etcd-2 is sending non stop traffic to etcd-1 ... These packets look like this inside etcd-1:
```
07:36:42.284441 IP (tos 0x0, ttl 62, id 64766, offset 0, flags [DF], proto TCP (6), length 60)
    10.42.93.129.60112 > 10.42.219.169.2380: Flags [S], cksum 0x4e4d (correct), seq 1210032937, win 27240, options [mss 1362,sackOK,TS val 33185998 ecr 0,nop,wscale 7], length 0
        0x0000:  0232 56c2 021b 0232 56a6 b6e2 0800 4500  .2V....2V.....E.
        0x0010:  003c fcfe 4000 3e06 f23e 0a2a 5d81 0a2a  .<..@.>..>.*]..*
        0x0020:  dba9 ead0 094c 481f a329 0000 0000 a002  .....LH..)......
        0x0030:  6a68 4e4d 0000 0204 0552 0402 080a 01fa  jhNM.....R......
        0x0040:  60ce 0000 0000 0103 0307
```

So even though the packets from IPSec container on host1 are with the current MAC address, the ARP entry corresponding to `10.42.93.129` is being kept alive incorrectly.

**Comment by leodotcloud** [comment]
This ARP issue between etcd seems to be quite easily reproducible with multiple force upgrades of both ipsec/ipsec-router containers.

**Comment by leodotcloud** [comment]
Seems to be a kernel issue to me, need to check if there is any issue reported already on the kernel mailing list.
```
root@soumyaadkubetest1:~# uname -a
Linux soumyaadkubetest1 4.4.0-93-generic #116-Ubuntu SMP Fri Aug 11 21:17:51 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
root@soumyaadkubetest1:~#
```

**Comment by deniseschannon** [comment]
`network-manager:v0.7.13` and `rancher/net:v0.13.1` are available in the `v1.6-development` branch

**Comment by galal-hussein** [comment]
After Rancher server upgrade from 1.6.10 to 1.6.11-rc4, and etcd was operational after upgrading ipsec and network services, also after force upgrading etcd multiple times on 6 hosts, etcd was able to start correctly without problems.
